docs/events_by_day.md

```diff
@@ -0,0 +1,113 @@
+* Memory usage of these streams can be very large due to both the number of messages and the verbosity of each message.  This can cause issues as Redis requires all data to be held in memory.
+
+To address these issues we suggest splitting each jobset into a logical stream per day.  This should enable old messages to be cleaned up, even if the jobset itself lasts indefinitely. It should also enable old, but still readbale, streams to be compressed which should further save memory.
+
+## Proposed Mechanism for Splitting Streams.
+
+### Write-Side
+
+* When an event message is received from Pulsar, the Pulsar message timestamp will be used to determine the (UTC) day to which the message pertains
+* The event message will then be appended to the redis stream under the key `<queue>_<jobset>_<day>` where `<day>` is the UTC day determined in the above step in hh_mm_ss format
+* The `<queue>_<jobset>_<day>` key that was appended to in the step above is then added to a Redis set stored at `<queue>_<jobset>_streams`
+* The result of the above should be that the messages for a a given jobset should be stored in one key per day. 
+The  sets stored under `<queue>_<jobset>_streams` can be used to determine which streams are available.  Because of the way we process pulsar messages, we can guarentee that 
+if a subsequent stream exits, any previous streams will never be appended to.
+
+### Read-Side
+* When a request for events is received, we first try to retrieve the set of streams at `<queue>_<jobset>_streams`.
+* If nothing exists under that key, we continue to poll until something becomes available.
+* Once `<queue>_<jobset>_<day>` is available, we first order the keys lexiographically and then call `XRevRangeN` to retrieve the latest MessageId for each stream.
+* We now identify the first stream that we need to read from.  In the common case where the user wants to read fromt eh start of the stream, this will simply 
+be the first lexiographically.  In the case where the user has provided an offset, we wil need to use the results of `XRevRangeN` determined in the previous step.
+* We continue to read the initial stream until we reach the value determined by `XRevRangeN`.  If the set we retrieved in step 1 indicates that there are more streams
+available then we can proceed to the next stream.  If this is the last stream **and** the current local time indicates that we are on the same utc day then 
+we should continue to listen for new events on this stream.   If the local time indicates that we are on a subsequent utc day, we should refetch `<queue>_<jobset>_streams` 
+and continue to poll the existing stream until this indicates that a new stream is available.
+
+### Considerations
+  
+We may have to repalce the autoexpiration of keys with a manual job to clean things up.  That is because when a stream `<queue>_<jobset>_<day>` is deleted,
```

Comment:
```suggestion
We may have to replace the autoexpiration of keys with a manual job to clean things up.  That is because when a stream `<queue>_<jobset>_<day>` is deleted,
```